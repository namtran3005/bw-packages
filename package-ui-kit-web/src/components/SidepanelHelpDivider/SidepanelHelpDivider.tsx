import * as React from 'react';
import styled from 'styled-components';
import { Spacing, Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

interface SidepanelHelpDividerProps {
  background?: ColorsType;
  backgroundAlpha?: number;
}

const getDividerColor = ({
  background = 'primaryLilac',
  backgroundAlpha,
}: SidepanelHelpDividerProps) => {
  if (backgroundAlpha) {
    return Colors.getWithAlpha(background, backgroundAlpha);
  } else {
    return Colors.get(background);
  }
};

const SidepanelHelpDivider: React.FC = styled('hr').withConfig({
  shouldForwardProp: (prop) =>
    !['background', 'backgroundAlpha'].includes(prop),
})<SidepanelHelpDividerProps>`
  height: ${Spacing.unit / 2}px;
  width: ${Spacing.base}px;
  background-color: ${getDividerColor};
  margin: ${Spacing.small}px 0;
  border: none;
`;

export { SidepanelHelpDivider, SidepanelHelpDividerProps };
