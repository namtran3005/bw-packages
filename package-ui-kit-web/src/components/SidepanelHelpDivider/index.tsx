export {
  SidepanelHelpDivider,
  SidepanelHelpDividerProps,
} from './SidepanelHelpDivider';
