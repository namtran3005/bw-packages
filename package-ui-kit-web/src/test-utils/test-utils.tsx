import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
