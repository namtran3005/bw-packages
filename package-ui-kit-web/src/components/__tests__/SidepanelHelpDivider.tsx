import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { SidepanelHelpDivider } from '../SidepanelHelpDivider';

describe('SidepanelHelpDivider', () => {
  it('should render divider', () => {
    const { baseElement } = renderWithTestHarness(<SidepanelHelpDivider />);
    expect(baseElement).toMatchSnapshot();
  });
});
