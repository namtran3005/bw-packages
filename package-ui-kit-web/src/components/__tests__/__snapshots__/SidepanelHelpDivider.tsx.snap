// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidepanelHelpDivider should render divider 1`] = `
.c0 {
  height: 4px;
  width: 24px;
  background-color: #BEAAFF;
  margin: 16px 0;
  border: none;
}

<body>
  <div>
    <hr
      class="c0"
    />
  </div>
</body>
`;
